action/protocol/genericvalidator.go

```diff
@@ -0,0 +1,63 @@
+package protocol
+
+import (
+	"context"
+	"github.com/pkg/errors"
+
+	"github.com/iotexproject/iotex-core/action"
+	"github.com/iotexproject/iotex-core/iotxaddress"
+)
+
+// GasLimit is the total gas limit could be consumed in a block
+const GasLimit = uint64(1000000000)
+
+// GenericValidator is the validator for generic action verification
+type GenericValidator struct{ cm ChainManager }
+
+// NewGenericValidator constructs a new genericValidator
+func NewGenericValidator(cm ChainManager) *GenericValidator { return &GenericValidator{cm} }
+
+// Validate validates a generic action
+func (v *GenericValidator) Validate(ctx context.Context, act action.Action) error {
+	vaCtx, validateInBlock := GetValidateActionsCtx(ctx)
+	if validateInBlock && (vaCtx.BlockHeight == 0 || act.SrcAddr() == "") {
```

Comment:
height == 0 is to skip validate for genesis block?
why SrcAddr() == "" can also skip validation?